packages/core/src/request.ts

```diff
@@ -1,11 +1,33 @@
-import { Event, SentryRequest, Session } from '@sentry/types';
+import { Event, SdkInfo, SentryRequest, Session } from '@sentry/types';
+import { isString } from '@sentry/utils';
 
 import { API } from './api';
 
+/** Extract envelope header containing sdk metadata from the API object */
+function getSdkMetadataForEnvelopeHeader(api: API): Partial<SdkInfo> | undefined {
+  if (!api.metadata) {
+    return;
+  }
+
+  const { name, version } = api.metadata;
+  if (!isString(name) || !isString(version)) {
+    return;
+  }
+
+  return {
+    ...(isString(name) && { name }),
+    ...(isString(version) && { version }),
```

Comment:
Do we really need so many `isString`?!

I'd expect `eventToSentryRequest` to access fields directly like `api.metadata.sdk.name`, `api.metadata.sdk.version`; well, in fact, just `api.metadata.sdk`.

Reply:
I'm just being defensive, as the data constructed here goes out directly to Relay. I can simplify this if we want.